upup/pkg/fi/cloudup/vsphere/vsphere_cloud.go

```diff
@@ -36,6 +40,18 @@ func (c *VSphereCloud) ProviderID() fi.CloudProviderID {
 	return fi.CloudProviderVSphere
 }
 
+func NewVSphereCloud(spec *kops.ClusterSpec) (*VSphereCloud, error) {
+	server := spec.VSphereServer
+	datacenter := spec.VSphereDatacenter
+	cluster := spec.VSphereResourcePool
+	username := os.Getenv("VSPHERE_USERNAME")
+	password := os.Getenv("VSPHERE_PASSWORD")
+
```

Comment:
oh sorry I think I read your original comments wrong :)
Yes you are right we should check the emptiness.
Let me add the check.